src/main/java/com/cinema/shop/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,53 @@
+package com.cinema.shop.dao.impl;
+
+import com.cinema.shop.dao.MovieSessionDao;
+import com.cinema.shop.exceptions.DataProcessingException;
+import com.cinema.shop.lib.Dao;
+import com.cinema.shop.model.MovieSession;
+import com.cinema.shop.util.HibernateUtil;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> query =
+                    session.createQuery("FROM MovieSession WHERE movie_id = :movieId"
+                            + " AND showTime BETWEEN :start AND :end", MovieSession.class);
+            query.setParameter("movieId", movieId);
+            query.setParameter("start", date.atTime(LocalTime.MIN));
+            query.setParameter("end", date.atTime(LocalTime.MAX));
+            return query.getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't find available Movie Sessions", e);
+        }
+    }
+
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movieSession);
+            transaction.commit();
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert Movie Session entity", e);
```

Comment:
let's make message more informative and add some info about entity we were trying to save (same goes for other methods with input parameters, where we throw exception)